### ISSUE TITLE
Add Flow#distinctUntilChanged?

### ISSUE THREAD

**Issue by He-Pin** [akka/akka#32081]
Motivation:
I find this pattern in some of our internal codebase, which is more like `statefulMap(..).collect{case Some(x) => x}` and I find the Kotlin has Flow#distinctUntilChanged too.

```scala

  def distinctUtilChanged:Repr[Out] = ???

  def distinctUtilChanged(predicate:(/*old*/ Out, /*new*/ Out) => Boolean):Repr[Out] = ???
```

**Comment by patriknw** [comment]
That can be done rather easily with `statefulMapConcat`?

**Comment by He-Pin** [comment]
> That can be done rather easily with `statefulMapConcat`?

Yes, but still better to have this out of box, let's wait to see if this can accept more votes.
